app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt

```diff
@@ -277,6 +300,24 @@ class DeadlineDetailsActivity : AppCompatActivity() {
             state = deadline.state
             val title = deadline.title
             val description = deadline.description
+            val group = deadline.owner
+
+            // Get the group of the deadline, if it exists
+            // If the group is owned by the user, they can edit it
+            val groupText = if (group is UserOwned) {
+                getString(R.string.not_in_any_group)
+            } else {
+                val groupDeadline = groupViewModel.getGroup((group as GroupOwned).groupID)
+                // If the group of the deadline is not owned by the user,
+                // they can't modify it
+                if (!(groupViewModel.getOwnedGroups().values.contains(groupDeadline))) {
+                    val modifyButton =
+                        findViewById<ImageButton>(R.id.deadline_details_activity_modify)
+                    modifyButton.isClickable = false
+                    modifyButton.isVisible = false
+                }
+                getString(R.string.in_the_group_X, groupDeadline.name)
+            }
```

Comment:
ooof that's a very big lambda, can you not refractor it a little by makinf some little function ?
In my opinion small lambda are more readable